### PR TITLE
[MRESOLVER-480] Document latest changes

### PR DESCRIPTION
**Pull request by cstamas**
Those that may affect consumer projects.

Also, a small irrelevant (and harmless) change: remove the class `DependencyScopes` that got introduced in alpha-7, but got deprecated while working on alpha-8, where we made Resolver fully scope oblivious.

---

https://issues.apache.org/jira/browse/MRESOLVER-480

### CODE REVIEW

**Comment by cstamas** [comment]
@gnodet @slawekjaranowski any opinion about this?

File: src/site/markdown/upgrading-resolver.md [hunk 1]
```diff
@@ -19,13 +19,13 @@
 This page will collect quick guides for developers upgrading from one to
 another major version of Resolver.
 
-# Upgrading from 1.x to 2.x
+# Upgrading From 1.x to 2.x
```
Comment (gnodet) [inline src/site/markdown/upgrading-resolver.md:22]: Unwanted capitalisation

File: src/site/markdown/upgrading-resolver.md [hunk 2]
```diff
@@ -19,13 +19,13 @@
...
 
 Maven Resolver upcoming major version 2.x should be "smooth sailing", as long you
 do not depend (directly or indirectly) on **deprecated** classes from Resolver
 1.x line. Always use latest 1.x release to check for deprecated classes.
 
-## Session handling changes
+## Session Handling Changes
```
Comment (gnodet) [inline src/site/markdown/upgrading-resolver.md:28]: Unwanted capitalisation

File: src/site/markdown/upgrading-resolver.md [hunk 3]
```diff
@@ -40,10 +40,31 @@ What changed on surface:
...
 * do not use `DefaultRepositorySystemSession` default constructor anymore.
 * instead, use `RepositorySystem#createSessionBuilder` to create `SessionBuilder` and out of it `CloseableSession` instances.
 * handle sessions as resources: each created instance should be closed once finished their use.
 * session instances created by given `RepositorySystem` should be used only with that same instance.
 * to shallow-copy session instances (for alteration purposes) using existing `DefaultRepositorySystemSession` copy constructor is acceptable (this is what Mojos do).
 * to shallow-copy session instances but have new lifecycle as well, use `SessionBuilder#withRepositorySystemSession` on newly created builder instances.
+
+## Consumer Project Changes (if more needed)
+
+Maven Resolver 2.x now delegates multiple aspects fully to consumer project, itself remaining "oblivious" about
```
Comment (gnodet) [inline src/site/markdown/upgrading-resolver.md:53]: ` now fully delegates multiple aspects to the consumer project`

File: src/site/markdown/upgrading-resolver.md [hunk 4]
```diff
@@ -40,10 +40,31 @@ What changed on surface:
...
+them. This was done to lessen (or better, fully remove) any overlap of logic between Resolver, and it's major
```
Comment (gnodet) [inline src/site/markdown/upgrading-resolver.md:54]: `any overlap between the logic of the Resolver`

File: src/site/markdown/upgrading-resolver.md [hunk 5]
```diff
@@ -40,10 +40,31 @@ What changed on surface:
...
+* The `ArtifactType` default implementation is deprecated, should be provided by consumer project instead, Resolver 2.x should not provide implementations for it.
+* Class path generation (in NodeListGenerator class variations in `org.eclipse.aether.util.graph.visitor` package) and class path filtering (in `DependencyFilterUtils`) is deprecated. It is the consumer app, based on own artifact properties, that can deduce how to build class path (or module path, or anything alike).
+* Dependency Scopes are fully removed. For Resolver, they were always "just labels". Resolver is now fully unaware of scopes (and for "system" artifacts has a dedicated handler exposed on session).
+
+For users of Maven Resolver Provider module, these changes will have clean migration path, as all the replacements for 
+items from the list above are present in `org.apache.maven:maven-resolver-provider` module. Still, for binary 
+compatibility, all affected classes are in fact left in place, but deprecated. Consumer projects should update 
+accordingly, and if they do use any of deprecated classes from resolver (affected modules are maven-resolver-api
```
Comment (gnodet) [inline src/site/markdown/upgrading-resolver.md:64]: `if they do use any deprecated classes from the resolver`

File: src/site/markdown/upgrading-resolver.md [hunk 6]
```diff
@@ -40,10 +40,31 @@ What changed on surface:
...
+For users of Maven Resolver Provider module, these changes will have clean migration path, as all the replacements for 
+items from the list above are present in `org.apache.maven:maven-resolver-provider` module. Still, for binary 
+compatibility, all affected classes are in fact left in place, but deprecated. Consumer projects should update 
+accordingly, and if they do use any of deprecated classes from resolver (affected modules are maven-resolver-api 
+and maven-resolver-util), they should switch to their counterparts, that are now present in maven-resolver-provider 
+module.
+
+Note: Resolver 2.x still fully supports "legacy" (Maven3-like) functionality, and if existing consumer
+code base adapts only the session related changes, consumer project can enjoy new features (like HTTP/2 transport),
+while keep full compatibility, identical to Resolver 1.x release.
```
Comment (gnodet) [inline src/site/markdown/upgrading-resolver.md:70]: `keeping`

**Review by gnodet — CHANGES_REQUESTED** [review]

**Comment by jira-importer** [comment]
Resolve #1151

**Pull request merged**